### PR TITLE
[WIP] Tarball ingestion

### PR DESCRIPTION
**Pull request by siscia**
PR dependent on the upgrade of gtest to the 1.8 version and integration of gmock.

This PR is a very initial sketch on how the ingestion of tarball may works.

Start to run CI and first impressions.

### CODE REVIEW

**Comment by siscia** [comment]
The travis machine miss libarchive.

Try to understand where to install it.

File: .travis.yml [hunk 1]
```diff
@@ -28,9 +28,10 @@ addons:
       - valgrind
       - voms-dev
       - gdb
+      - libarchive-dev
 
 before_script:
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update && brew cask install osxfuse; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update && brew cask install osxfuse && brew install libarchive; fi
```
Comment (jblomer) [inline .travis.yml]: So... on travis macOS nodes libarchive is missing? Isn't that a default library that comes with macOS?

**Comment by siscia** [comment]
@jblomer , @radupopescu I was wrong, the .zip of libarchive is actually 7.4 MB, do I go ahead and integrate is as external library anyway?

**Comment by radupopescu** [comment]
@siscia @jblomer I looked at some numbers:

* CernVM-FS sources are 41.6 MB (download ZIP from GitHub).
* libarchive-3.3.2.tar.gz is 6 MB on disk
* libarchive.a is 1.2 MB

I wonder what is the size increase of linking libarchive.a with cvmfs_swissknife?

File: CMakeLists.txt [hunk 2]
```diff
@@ -314,6 +314,9 @@ set (INCLUDE_DIRECTORIES ${INCLUDE_DIRECTORIES} ${VJSON_INCLUDE_DIRS})
 find_package (LibArchive REQUIRED)
 set (INCLUDE_DIRECTORIES ${INCLUDE_DIRECTORIES} ${LibArchive_INCLUDE_DIRS})
 
+find_package (ICONV REQUIRED)
```
Comment (jblomer) [inline CMakeLists.txt]: Isn't libarchive told not to use iconv?

**Comment by jblomer** [comment]
Superseded by #2059 and #2060

**Pull request closed without merging**